overlord/snapstate/target.go

```diff
@@ -65,6 +65,9 @@ type Options struct {
 	// pre-existing behavior of calling InstallMany with one snap vs calling
 	// Install.
 	ExpectOneSnap bool
+	// NewModel is the new model in case the operation is part of a remodel
+	// change.
+	NewModel *asserts.Model
```

Comment:
we should use device contexts as designed and not do this, or have a discussion to be convinced otherwise because I don't really like the idea of having two ways of doing the same thing